Automattic/wp-calypso
# Checkout: Remove product_id requirement for sending products to cart

#### Changes proposed in this Pull Request

When sending products to the shopping cart endpoint, it is required that they have a `product_slug` and a `product_id`. However, since most of the code in calypso operates on the `product_slug`, it requires a lot of effort to add a product with an ID. Typically we have to query the products endpoint, find the matching product by slug, and then apply that product's ID. This also makes TypeScript types confusing because we need a type for the intermediate stage of having all required data _except_ for `product_id` and another type for the data _with_ `product_id`.

In D71913-code, we make `product_id` optional and rely on `product_slug` alone (`product_id` will still be added but the adding will occur inside the endpoint). 

This PR removes the requirement of `product_id` in the `@automattic/shopping-cart` package and updates many places in the code that previously added the property such that they no longer do.

#### Testing instructions

This touches many areas of the codebase by removing the `fillInSingleCartItemAttributes` function but (as long as there are no syntax errors) as long as one of them works, all of them should work, because the only thing that function did was to add a `product_id`.

- Apply D71913-code and sandbox the API.
- Add a product to your cart (we should test different products like new plans, renewals, domains, signup, etc.) and visit checkout.
- Verify that the product you added shows up correctly in checkout.
